### ISSUE TITLE
cannot open a jupytext notebook in nb7

### ISSUE THREAD

**Issue by parmentelat** [jupyter/notebook#6914]
**Describe the bug**

The jupytext extension does not work as expected in nb7

**To Reproduce**

1. install jlab4 as released in pip
2. install nb7 with `pip install -U --pre notebook`
3. install jupytext with `pip install jupytext`
4. install the jupytext labextension with
  `pip install git+https://github.com/mwouts/jupyterlab-jupytext.git@jupyterlab4`
5. create a simple jupytext notebook named for instance `foo-nb.md`
6. under jlab, check that you can open the notebook: right click → "Jupytext Notebook"
7. perform the same operation from the regular notebook tree browser: the "Jupytext Notebook" option is present but does not open the file as a notebook, it shows a text editor instead

**Expected behavior**
1. we should be able to open a jupytext notebook from the regular notebook file browser; 
2. (also when opening a URL like localhost:8888/notebooks/path/file.md we need this to open a notebook and not a text editor, but for that to work the former behaviour 1. must first work)

**Screenshots**

![image](https://github.com/jupyter/notebook/assets/4519227/3f764548-74e9-4ec5-9334-1af96299bad6)
 
**Desktop (please complete the following information):**

- OS: \[e.g. iOS\] MacOS & linux
- Browser \[e.g. chrome, safari\] - chrome
- Version \[e.g. 22\]

* jlab 4.0.2
* notebook 7.0.0b4
* as of this writing, https://github.com/mwouts/jupyterlab-jupytext.git@jupyterlab4 is on commit d451f5c
 
**Additional context**

Please see also https://discourse.jupyter.org/t/generic-question-about-jlab4-and-nb7/19846

**Comment by jtpio** [comment]
Thanks @parmentelat for reporting.



> Steps to reproduce the behavior:

Looks like we also need to install `jupytext` with `pip install jupytext`? Not sure `jupyterlab-jupytext` installs `jupytext` automatically?

After following the instructions above and the additional `pip install jupytext` it seems like it's possible to open the markdown file with the notebook factory via the context menu:

https://github.com/jupyter/notebook/assets/591645/392196d7-2961-456f-a79c-ab059f521b02

But the behavior seems to be similar in JupyterLab right? By default double-clicking on the markdown file or opening it with the default "Open" context menu shows the markdown source:

https://github.com/jupyter/notebook/assets/591645/f3127393-4e8b-4e3d-aee6-94313fd4ee48

**Comment by jtpio** [comment]
Also this might be related to https://github.com/jupyter/notebook/issues/6324.

**Comment by parmentelat** [comment]
so, I confirm the I had tried the 'Open with' menu when reporting this

however I must admit that I am juggling between many environments, switching between stable and unstable versions of many different components, this is a bit overwhelming and my envs are in shambles...
so I guess I must have screwed up somehow somewhere; plus, the branches may have changed meanwhile

I will restart from scratch when I have a chance, most likely since all worked well for you, I should be able to report that as a red herring

**Comment by jtpio** [comment]
ok no problem, thank you so much for trying Notebook 7!

**Comment by parmentelat** [comment]
also, probably relevant to this discussion, I must say that my environment includes the following, which is needed in the context of jlab to get the default action (e.g. double click) to open as a notebook

I might redo the experiment without that in place, if that sounds a useful test (although it should only help, in theory, no ?)

```
cat ~/.jupyter/labconfig/default_setting_overrides.json
{
  "@jupyterlab/docmanager-extension:plugin": {
      "defaultViewers": {
          "python": "Jupytext Notebook",
          "markdown": "Jupytext Notebook"
    }
  }
}
```

**Comment by jtpio** [comment]
Just checked and it seems like the default viewers are not respected in Notebook 7 while they are in JupyterLab:

https://github.com/jupyter/notebook/assets/591645/93a6fc4f-d6ed-4f41-81d5-ed5c29a7737b

**Comment by parmentelat** [comment]
Hiya lads 

I am currently struggling with a bug on our deployment when trying to use nb7
and it might be related to this issue

in a nutshell, I run something like this
```
jupyter lab  --NotebookApp.base_url=/some-container-dep-prefix/
```

and apparently when trying to open a jupytext notebook the app answers with a redirect to 
`/edit/the-path/to-the-notebook`
while I really need it to redirect to
`/some-container-dep-prefix/edit/the-path/to-the-notebook`

I just came across this and don't yet have an easily reproducible sandbox, 
but is it possible that a hasty change has forgotten to take this `base_url` setting into account ?

---
admittedly the relationship with this issue is just a hunch on my side,
so please let me know if you'd rather deal with this in a separate issue instead

**Comment by jtpio** [comment]
Thanks @parmentelat.

Yes it looks like an issue with the base URL.

FYI I submitted a fix for https://github.com/jupyter/notebook/issues/6940 in https://github.com/jupyter/notebook/pull/6941 which might be related.

This is available in `7.0.0rc1` if you would like to try it: https://github.com/jupyter/notebook/releases/tag/v7.0.0rc1

Otherwise yes a new issue to track this would be great, thanks!
